functions/api/index.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response } from 'express'
+import { ApolloServer } from 'apollo-server-cloud-functions'
+import { buildSchema } from 'type-graphql'
+import { HelloWorldResolver } from './resolvers'
+
+export const apiServer = async (req: Request, res: Response) => {
```

Comment:
You don't need `async` here, and in fact, if you returned a promise this would stop working. You can see UI for an example for the hack I need to use for async initialization.

Reply:
I do for `  const schema = await buildSchema({
    resolvers: [HelloWorldResolver],
  })`

Reply:
This function is a replication of your async hack. I just do it in a separate file instead of within the `functions/index.ts`

Reply:
Oh, whoops, missed that one. Actually, I was wrong about this comment. It's not async on initialization, it initializes Apollo for every request. That makes sense for functions I guess, since they are ephemeral. Though they do get reused sometimes. I wonder what the perf difference would be there. 🤔

Reply:
`export const api = https.onRequest(apiServer)`
Or 
```ts
export const api = https.onRequest(async (req, res) => {
  const schema = await buildSchema({
    resolvers: [HelloWorldResolver],
  })

  if (req.url === '/api') req.url = '/';
  
  const apolloServer = new ApolloServer({
    schema,
    introspection: true,
    playground: false
  })
  return apolloServer.createHandler({ cors: { origin: '*' }})(req, res)
})
```
Are both the same thing

Reply:
I've considered generating the schema outside of function and reusing since we can't modify schema once it's deployed.